# Eksplorasi Breast Cancer Wisconsin Diagnostic Dataset

Notebook ini digunakan untuk memahami struktur Breast Cancer Wisconsin Diagnostic Dataset sebelum dilakukan eksperimen Machine Learning.

Tujuan:
- Memuat dataset.
- Melihat ukuran dan struktur data.
- Memahami fitur dan label.
- Memisahkan fitur (`X`) dan label (`y`).
- Memeriksa distribusi kelas.
- Memeriksa missing value.

In [38]:
import pandas as pd

from sklearn.datasets import load_breast_cancer

## 1. Memuat Dataset

Breast Cancer Wisconsin Diagnostic Dataset tersedia secara langsung melalui library scikit-learn.

Dataset berisi sejumlah fitur numerik yang berasal dari karakteristik sel dan digunakan untuk membedakan dua kelas.

In [39]:
data = load_breast_cancer()

type(data)
df = pd.DataFrame(
    data.data,
    columns=data.feature_names
)

df["target"] = data.target

df.head()

,mean radius,mean texture,mean perimeter,mean area,mean smoothness,mean compactness,mean concavity,mean concave points,mean symmetry,mean fractal dimension,...,worst texture,worst perimeter,worst area,worst smoothness,worst compactness,worst concavity,worst concave points,worst symmetry,worst fractal dimension,target
0,17.99,10.38,122.80,1001.0,0.11840,0.27760,0.3001,0.14710,0.2419,0.07871,...,17.33,184.60,2019.0,0.1622,0.6656,0.7119,0.2654,0.4601,0.11890,0
1,20.57,17.77,132.90,1326.0,0.08474,0.07864,0.0869,0.07017,0.1812,0.05667,...,23.41,158.80,1956.0,0.1238,0.1866,0.2416,0.1860,0.2750,0.08902,0
2,19.69,21.25,130.00,1203.0,0.10960,0.15990,0.1974,0.12790,0.2069,0.05999,...,25.53,152.50,1709.0,0.1444,0.4245,0.4504,0.2430,0.3613,0.08758,0
3,11.42,20.38,77.58,386.1,0.14250,0.28390,0.2414,0.10520,0.2597,0.09744,...,26.50,98.87,567.7,0.2098,0.8663,0.6869,0.2575,0.6638,0.17300,0
4,20.29,14.34,135.10,1297.0,0.10030,0.13280,0.1980,0.10430,0.1809,0.05883,...,16.67,152.20,1575.0,0.1374,0.2050,0.4000,0.1625,0.2364,0.07678,0


## 2. Ukuran Dataset

Ukuran dataset diperiksa untuk mengetahui jumlah sampel dan jumlah variabel yang tersedia.

In [40]:
df.shape

(569, 31)

## 3. Memisahkan Fitur dan Label

Fitur (`X`) berisi 30 variabel numerik yang digunakan oleh model untuk melakukan klasifikasi.

Label (`y`) berisi kelas sebenarnya dari setiap sampel.

In [41]:
X = df.drop("target", axis=1)
y = df["target"]

In [42]:
X.head()

,mean radius,mean texture,mean perimeter,mean area,mean smoothness,mean compactness,mean concavity,mean concave points,mean symmetry,mean fractal dimension,...,worst radius,worst texture,worst perimeter,worst area,worst smoothness,worst compactness,worst concavity,worst concave points,worst symmetry,worst fractal dimension
0,17.99,10.38,122.80,1001.0,0.11840,0.27760,0.3001,0.14710,0.2419,0.07871,...,25.38,17.33,184.60,2019.0,0.1622,0.6656,0.7119,0.2654,0.4601,0.11890
1,20.57,17.77,132.90,1326.0,0.08474,0.07864,0.0869,0.07017,0.1812,0.05667,...,24.99,23.41,158.80,1956.0,0.1238,0.1866,0.2416,0.1860,0.2750,0.08902
2,19.69,21.25,130.00,1203.0,0.10960,0.15990,0.1974,0.12790,0.2069,0.05999,...,23.57,25.53,152.50,1709.0,0.1444,0.4245,0.4504,0.2430,0.3613,0.08758
3,11.42,20.38,77.58,386.1,0.14250,0.28390,0.2414,0.10520,0.2597,0.09744,...,14.91,26.50,98.87,567.7,0.2098,0.8663,0.6869,0.2575,0.6638,0.17300
4,20.29,14.34,135.10,1297.0,0.10030,0.13280,0.1980,0.10430,0.1809,0.05883,...,22.54,16.67,152.20,1575.0,0.1374,0.2050,0.4000,0.1625,0.2364,0.07678


In [43]:
y.head()

0    0
1    0
2    0
3    0
4    0
Name: target, dtype: int64

In [44]:
data.target_names

array(['malignant', 'benign'], dtype='<U9')

## 4. Distribusi Kelas

Distribusi kelas diperiksa untuk mengetahui jumlah data pada masing-masing kelas.

In [45]:
y.value_counts().sort_index()

target
0    212
1    357
Name: count, dtype: int64

In [46]:
print("Malignant :", (y == 0).sum())
print("Benign    :", (y == 1).sum())

Malignant : 212
Benign    : 357


## 5. Pemeriksaan Missing Value

Missing value diperiksa karena data kosong dapat memengaruhi proses training dan evaluasi model.

In [47]:
df.isnull().sum().sum()

np.int64(0)

## Kesimpulan Eksplorasi Awal

Breast Cancer Wisconsin Diagnostic Dataset memiliki 569 sampel dengan 30 fitur numerik dan dua kelas.

Dataset tidak memiliki missing value dan dapat digunakan untuk eksperimen klasifikasi menggunakan Decision Tree, KNN, dan Random Forest.

Eksperimen berikutnya akan menggunakan desain yang sama seperti pada Dataset Wine, yaitu variasi random seed pada pembagian data training dan testing.